### PR TITLE
Add FabricIndex, extract data model, new spec ref

### PR DESCRIPTION
**Pull request by mfucci**

### CODE REVIEW

**Comment by mfucci** [comment]
Include a proposal to cleanup our references to Matter specs.

File: src/Specifications.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** {@link https://csa-iot.org/developer-resource/specifications-download-request/ Matter Core Specification 1.0} */
```
Comment (Apollon77) [inline src/Specifications.ts:7]: Is that really the URL?

File: src/matter/common/FabricIndex.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Each fabric supported on a node is referenced by fabric-index that is unique on the node. This
+ * fab­ric-index enables the look-up of the full fabric information from the fabric-index. A fabric-index
+ * of 0 (zero) or null SHALL indicate that there is no fabric associated with the context in which the
+ * fab­ric-index is being used. If fabric-index is used in a context that is exclusively associated with
+ * a fab­ric, such as fabric-scoped data model elements, then the fabric-index values SHALL NOT include 0
+ * (zero) or null.
+ * 
+ * @see [Matter Specification R1.0], section 7.5.2
```
Comment (Apollon77) [inline src/matter/common/FabricIndex.ts:17]: "section" or § as above? We should decide for one ;-)

**Review by Apollon77 — DISMISSED** [review]

File: src/Specifications.ts (hunk 1 above)
Comment (mfucci) [reply src/Specifications.ts:7]: Yes, that's the most direct link to download the specs. The specs themselves don't have a direct URL...

File: src/matter/common/FabricIndex.ts (hunk 2 above)
Comment (mfucci) [reply src/matter/common/FabricIndex.ts]: Done!

File: src/Specifications.ts (hunk 1 above)
Comment (Apollon77) [reply src/Specifications.ts:7]: Ahh, ok so read as This link and then document X ... got it

**Comment by mfucci** [comment]
Update spec references!

File: src/codec/TlvObjectCodec.ts [hunk 3]
```diff
@@ -20,6 +20,10 @@ export interface Field<T> extends TaggedTemplate<T> {optional: false};
 export interface OptionalField<T> extends TaggedTemplate<T> {optional: true};
 export type FieldTemplates = {[key: string]: Field<any> | OptionalField<any>};
 
+interface BitTemplate { position: number }
```
Comment (Apollon77) [inline src/codec/TlvObjectCodec.ts:23]: The added Butmap support by fault or intent? (I have no issue adding it, but formally different topic ... we can be pragmatic :-)

**Review by Apollon77 — DISMISSED** [review]

File: src/codec/TlvObjectCodec.ts (hunk 3 above)
Comment (mfucci) [reply src/codec/TlvObjectCodec.ts:23]: Oups, by fault. I will move to another PR

**Review by Apollon77 — APPROVED** [review]

**Pull request merged**